src/modules/accounts/useCases/RegisterUser/RegisterUserController.ts

```diff
@@ -27,7 +30,14 @@ export class RegisterUserController implements Controller {
     password_confirmation,
   }: RegisterUserControllerRequest): Promise<HttpResponse> {
     try {
-      // TODO: Add validation
+      const validation = this.validator.validate({
```

Comment:
I think that here we can refactor like:
```js
async handle(request: RegisterUserControllerRequest): Promise<HttpResponse> {
  const validationResult = this.validator.validate(data);
  
  if (validationResult.isLeft()) {
    return clientError(validation.value);
  }
  
  const { name, email, password, password_confirmation } = request;
  
  ...
}
```
That's because the validator will always receive the full request (I think).

Reply:
Yes, the idea is that the validator will always receive the entire request body (due this we can pass the request interface as a generic type to the validator). Ok, perfect, this refactor wont affect the validation behavior. I only passed on the properties to the validator because the they were already unstructured.

Reply:
Yeah, I got it, I send this proposal of refactor so, when we add a new parameter to the request, we won't need to type it twice inside the validator and the use case call.